src/hooks/useSoftkey.js

```diff
@@ -1,4 +1,21 @@
-import { useContext } from 'preact/hooks'
+import { useContext, useEffect } from 'preact/hooks'
 import { SoftkeyContext } from 'contexts'
 
-export const useSoftkey = () => useContext(SoftkeyContext)
+export const useSoftkey = (origin, newContext, config = null, dependencies = []) => {
+  const softkey = useContext(SoftkeyContext)
+  // todo: push the stack concept all the way down to useKeys
+  // so that other handlers (useNavigation) don't work while
+  // a popup is open
+  useEffect(() => {
+    console.log('useSoftkey', origin, newContext)
+    newContext && softkey.dispatch({ type: 'push' })
+    return () => { newContext && softkey.dispatch({ type: 'pop' }) }
+  }, [newContext])
+
+  useEffect(() => {
+    if (config) {
+      softkey.dispatch({ type: 'set', config })
+    }
+  }, dependencies)
+  return softkey
+}
```

Comment:
route to a new page, popup a new component, always `newContext=true`

in which case do you think the `newContext` needed to be `false` here?

Reply:
In this commit, it is `false` in `<ArticleSection>` to signify to it contributes to the context created by `<Article>` and doesn't replace it. 

Another way to achieve something similar would be to use a string like the `origin` param to specify which view a component belongs to. All components of the article view could specify `article` to share a context.

Reply:
The second option proposed here has the advantage of being independent from the call order.

Reply:
👍 I see the latest commit, `origin` looks clearer to define which context it is.